src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,801 @@
+                    break;
+                case EndFW.TYPE_ID:
+                    final EndFW end = endRO.wrap(buffer, index, index + length);
+                    onApplicationEnd(end);
+                    break;
+                case WindowFW.TYPE_ID:
+                    final WindowFW window = windowRO.wrap(buffer, index, index + length);
+                    onApplicationWindow(window);
+                    break;
+                case ResetFW.TYPE_ID:
+                    final ResetFW reset = resetRO.wrap(buffer, index, index + length);
+                    onApplicationReset(reset);
+                    break;
+            }
+        }
+
+        private void onApplicationWindow(
+            WindowFW window)
+        {
+            //TODO
+        }
+
+        private void onApplicationReset(
+            ResetFW reset)
+        {
+            //TODO
+        }
+
+        private void onApplicationBegin(
+            BeginFW begin)
+        {
+
+            OctetsFW extension = begin.extension();
+            SocksBeginExFW socksBeginEx = extension.get(socksBeginExRO::wrap);
+
+            String address = socksBeginEx.address().asString();
+            int port = socksBeginEx.port();
+
+            this.receiver.doSocksCommandReply(address, port);
+
+        }
+
+        private void onApplicationData(
+            DataFW data)
+        {
+            final OctetsFW payload = data.payload();
+            //TODO
+        }
+
+        private void onApplicationEnd(
+            EndFW end)
+        {
+            final long traceId = end.trace();
+            doApplicationEnd(traceId);
+        }
+
+        private void doApplicationBegin(
+            MessageConsumer target,
+            long streamId,
+            long traceId,
+            String address,
+            int port)
+        {
+
+            SocksBeginExFW socksBeginEx = socksBeginExRW.wrap(extBuffer, 0, extBuffer.capacity())
+                .typeId(socksTypeId)
+                .address(address)
+                .port(port)
+                .build();
+
+            final BeginFW begin = beginRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                                         .routeId(this.routeId)
+                                         .streamId(streamId)
+                                         .trace(traceId)
+                                         .extension(socksBeginEx.buffer(), socksBeginEx.offset(), socksBeginEx.sizeof())
+                                         .build();
+
+            target.accept(begin.typeId(), begin.buffer(), begin.offset(), begin.limit());
+        }
+
+        private void doApplicationData(
+            DirectBuffer buffer,
+            int offset,
+            int sizeOf)
+        {
+            final DataFW data = dataRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                                      .routeId(routeId)
+                                      .streamId(replyId)
```

Comment:
Should this be `replyId` or `initialId`?
Please review all `doApplication*` methods for consistency.